syntax="proto3";

service WorkerServer {
    rpc ProcessParagraph(ParagraphRequest) returns (ParagraphResponse) {}
    rpc Heartbeat(StatusRequest) returns (StatusResponse) {}
}

message ParagraphRequest {
    string formation_name = 1;
    string paper_id = 2;
    string paragraph = 3;
} 

message Triplet {
    string head = 1;
    string tail = 2;
    string relationship_type = 3;
}

message ParagraphResponse {
    bool status = 1;
    repeated Triplet relationships = 2;
    string model_used = 3;
} 

message ErrorResponse {
    string error = 1;
}

message StatusRequest {}

message StatusResponse {
    bool status = 1;
}